### DISCUSSION TITLE
git bash symbolic links on windows

### DISCUSSION THREAD

**Discussion by bistritapcv** [community/community#23591] [New to GitHub]
<p>I did an ln -s file1 file2  in gitbash.</p>
<p>when I did an ls -l file2 it did not have an “s” in it.</p>
<p>When I did “stat” on each file, there were different inodes.</p>
<p>I was wondering whether “ln -s” (symbolic links) do not work on git bash in Windows 7?</p>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/bistritapcv">@bistritapcv</a>,</p>
<p>This post was moved to a different board that fits your topic of discussion a bit better. This means you’ll get better engagement on your post, and it keeps our Community organized so users can more easily find information.</p>
<p>As you’ll notice, your Topic is now in the How to use Git and GitHub board. No action is needed on your part; you can continue the conversation as normal here.</p>
<p>Let me know if you have any other questions or if I can help with anything else.</p>
<p>Cheers!</p>

**Comment by bistritapcv** [comment]
<p>Unfortunately, no one has answered.</p>

**Comment by primetheus** [comment, marked as answer]
<p>Symbolic links on Git Bash for Windows needs to be enabled when you install it</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/271993b6cebb4de88b38a4ca652d7f701c5d4ce2.png" data-download-href="/uploads/short-url/5zTq1QnY33HmqqQdYrzQ1mg9Bl0.png?dl=1" title="1LKsG.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/865381/181078107-553a4a63-2b14-4c4c-a0bd-e38ffe696bb7.png" alt="1LKsG.png" data-base62-sha1="5zTq1QnY33HmqqQdYrzQ1mg9Bl0" width="601" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/271993b6cebb4de88b38a4ca652d7f701c5d4ce2_2_601x500.png, https://aws1.discourse-cdn.com/github/original/2X/2/271993b6cebb4de88b38a4ca652d7f701c5d4ce2.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/2/271993b6cebb4de88b38a4ca652d7f701c5d4ce2.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/271993b6cebb4de88b38a4ca652d7f701c5d4ce2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">1LKsG.png</span><span class="informations">747×621 106 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by gani247** [comment]
<p>how to get it enabled after its installed</p>

**Comment by Zhen-hui** [comment]
<p>I would just reinstall gitbash, quick and easy!</p>

**Comment by CapinWinky** [comment]
<p>You can set this in the config file for git for windows. By default, this is located at <code>C:\ProgramData\Git\config</code>.  </p>
<p>Find the <code>[core]</code> section and add <code>symlinks = true</code> to turn it on. </p>

**Reply by GabrielChavesBorges** [reply]
In my PC the path was "C:\Program Files\Git\etc" and the file name was "gitconfig".

**Reply by michael-ansolis** [reply]
On my PC, the path was C:\ProgramData\Git\config
The parameter in question was "symlinks" in the [core] section.

**Comment by vpArth** [comment]
<p>Hm, I have <code>symlinks=true</code>, but <code>ln -s largeFolder shortcut</code> just starts to make full copy… </p>
<pre><code>$ cat /c/ProgramData/Git/config
[core]
symlinks = true

$ echo Orig &gt; 1
$ ln -s 1 2
$ cat 2
Orig
$ echo Modified &gt; 2
$ head 1 2
==&gt; 1 &lt;==
Orig
==&gt; 2 &lt;==
Modified
</code></pre>

**Comment by s-weigand** [comment]
<p>This answer worked for me:<br>
<a href="https://stackoverflow.com/a/40914277/3990615" rel="nofollow noopener">https://stackoverflow.com/a/40914277/3990615</a></p>

**Reply by lost-scripts** [reply]
And, in my case at least, more concretely Xavi Montero comment's [**link**](https://stackoverflow.com/questions/58857724/how-can-i-make-symlinks-made-from-inside-docker-linux-containers-to-be-seen-from/63325536#63325536) there to a related question's own answer 🙂

**Comment by jminne** [comment]
<p>That did it!</p>
<ol>
<li>
<p>run as admin</p>
</li>
<li>
<p>export MSYS=winsymlinks:nativestrict</p>
</li>
</ol>

**Reply by cglacet** [reply]
What do you run as admin?

**Reply by yoderj** [reply]
See [my answer](https://github.com/orgs/community/discussions/23591#discussioncomment-3241027):

> That is, run Git Bash as admin, run the command export MSYS=winsymlinks:nativestrict, then run ln -s oldFile newLink to your hearts content within that session.

Though [jverne](https://github.com/orgs/community/discussions/23591#discussioncomment-3241026) notes:

> If you enable “Developer mode” in the Windows 10 settings you don’t have to run the shell as Admin.

**Comment by jverne** [comment]
<p>If you enable “Developer mode” in the Windows 10 settings you don’t have to run the shell as Admin.</p>

**Comment by yoderj** [comment]
<p>That is, run Git Bash as admin, run the command <code>export MSYS=winsymlinks:nativestrict</code>, then run <code>ln -s oldFile newLink</code> to your hearts content within that session.</p>

**Comment by CervEdin** [comment]
<p>I can confirm <code>ln</code> symbolic link behavior is determined by the <code>MSYS</code> environment variable, more info the <a href="https://cygwin.com/cygwin-ug-net/using.html#pathnames-symlinks" rel="noopener nofollow ugc">Cygwin docs</a>.</p>
<p>You don’t need to be admin, just <a href="https://docs.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/create-symbolic-links" rel="noopener nofollow ugc">configure the <code>User Rights Assignment</code></a></p>

**Comment by oleksis** [comment]
<p>Before clone the repository you can set the config:</p>
<pre><code class="lang-bash">git config --global core.symlinks true
</code></pre>

**Comment by hargobind** [comment]
If you add the `MSYS` variable to your system's environment variables, you won't have to run `export MSYS...` every time.

See [Setting the path and variables in Windows](https://www.computerhope.com/issues/ch000549.htm)

Variable name = `MSYS`
Variable value = `winsymlinks:nativestrict`

Now every time you open your bash shell, the setting will be there.

**Comment by levidavidmurray** [comment]
Jesus. Took me like 30 minutes to figure out the inconsistencies I was facing was because `ln -s` created a copy instead of a symlink. Very strange default behaviour.
